packages/ui/src/components/Visualization/Canvas/FormV2/hooks/one-of-field.ts

```diff
@@ -25,27 +25,30 @@ export const useOneOfField = (propName: string) => {
       return;
     }
 
-    if (typeof value === 'object' && isDefined(selectedOneOfSchema?.schema.properties)) {
-      const newValue = { ...value };
-      Object.keys(selectedOneOfSchema.schema.properties).forEach((prop) => {
-        delete (newValue as Record<string, unknown>)[prop];
+    const newValue = { ...value };
+    if (typeof value === 'object') {
+      const oldProps = selectedOneOfSchema?.schema.properties;
+      if (isDefined(oldProps)) {
+        Object.keys(oldProps).forEach((prop) => delete (newValue as Record<string, unknown>)[prop]);
+      }
+    }
+
+    const newProps = schema?.schema.properties;
+    if (isDefined(newProps)) {
+      Object.keys(newProps).forEach((prop) => {
+        setValue(newValue, prop, {});
       });
```

Comment:
We're assuming we're gonna always render objects, but there's also the chance the prop is a string